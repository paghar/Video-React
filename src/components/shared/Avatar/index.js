//@flow
import React from 'react';
import { AvatarBtn } from './index.style';
import Typography from 'components/shared/Typography';

type Props = {
  backgroundColor: string,
  imgUrl?: string,
  isBorder?: boolean,
  children: string,
  width: string,
  height: string,
  borderRadius: string,
  font: Object,
  color: string,
  avatarClick?: () => void,
  className?: string,
};

function Avatar({
  backgroundColor,
  imgUrl,
  isBorder,
  children,
  width,
  height,
  borderRadius,
  font,
  color,
  avatarClick,
  className,
}: Props) {
  return (
    <AvatarBtn
      backgroundColor={backgroundColor}
      imgUrl={imgUrl}
      isBorder={isBorder}
      width={width}
      height={height}
      borderRadius={borderRadius}
      onClick={avatarClick}
      className={className}
    >
      {!imgUrl && (
        <Typography font={font} color={color} hoverable>
          {children}
        </Typography>
      )}
    </AvatarBtn>
  );
}

export default Avatar;
