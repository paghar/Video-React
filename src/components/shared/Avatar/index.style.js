//@flow
import styled from 'styled-components';

export const AvatarBtn = styled.button`
  background-color:${({ backgroundColor }) => backgroundColor}  
  ${({ imgUrl }) => imgUrl && `background-image: url(${imgUrl});`}
  background-size: 100% 100%;
   width: ${({ width }) => width};
  height: ${({ height }) => height};
  border-radius: ${({ borderRadius }) => borderRadius};
  border: ${({ isBorder }) => (isBorder ? '2px' : '0px')} solid #0d0d1a;  
  display: flex;
  justify-content: center;
  align-items: center;
  outline: none;
`;
