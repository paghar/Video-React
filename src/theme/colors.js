import theme from 'styled-theming';

export const Generals = {
  i: theme('mode', {
    dark: '#05050d',
  }),
  ii: theme('mode', {
    dark: '#0d0d1a',
  }),
  iii: theme('mode', {
    dark: '#151526',
  }),
  iv: theme('mode', {
    dark: '#212133',
  }),
  v: theme('mode', {
    dark: '#262640',
  }),
  vi: theme('mode', {
    dark: '#7b7b9a',
  }),
  vii: theme('mode', {
    dark: '#b7b7cc',
  }),
  viii: theme('mode', {
    dark: '#ffffff',
  }),
  x: theme('mode', {
    dark: '#e6e6f3',
  }),
  ix: theme('mode', {
    dark: '#f1f1f9',
  }),
};

export const Primary = {
  Dim: theme('mode', {
    dark: '#2172ff',
  }),
  Light: theme('mode', {
    dark: '#317BFC',
  }),
  Dark: theme('mode', {
    dark: '#0E5EE8',
  }),
};

export const PromptWarn = {
  Dim: theme('mode', {
    dark: '#F51653',
  }),
  Light: theme('mode', {
    dark: '#FF4074',
  }),
  Dark: theme('mode', {
    dark: '#DE0540',
  }),
};

export const Secondary = {
  i: theme('mode', {
    dark: '#00d3b2',
  }),
  ii: theme('mode', {
    dark: '#ffcc00',
  }),
  iii: theme('mode', {
    dark: '#f51653',
  }),
  iv: theme('mode', {
    dark: '#64a7ff',
  }),
};

export const Avatars = [
  theme('mode', {
    dark: '#5c4dff',
  }),
  theme('mode', {
    dark: '#685ce6',
  }),
  theme('mode', {
    dark: '#7d73e6',
  }),
  theme('mode', {
    dark: '#928ae6',
  }),
];

export const Gradient = theme('mode', {
  dark:
    'linear-gradient(to bottom, rgba(21, 21, 38, 0), rgba(21, 21, 38, 0.69) 53%, #111120)',
});
