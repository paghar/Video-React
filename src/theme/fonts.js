import theme from 'styled-theming';

export const Title1 = {
  fontName: theme('size', {
    medium: "'Red Hat Text', sans-serif",
  }),
  fontSize: theme('size', {
    medium: '22px',
  }),
  fontWeight: theme('size', {
    medium: '400',
  }),
  lineHeight: theme('size', {
    medium: '35px',
  }),
};

export const Title2 = {
  fontName: theme('size', {
    medium: "'Red Hat Text', sans-serif",
  }),
  fontSize: theme('size', {
    medium: '21px',
  }),
  fontWeight: theme('size', {
    medium: '500',
  }),
  lineHeight: theme('size', {
    medium: '33px',
  }),
};

export const Display1 = {
  fontName: theme('size', {
    medium: "'Red Hat Text', sans-serif",
  }),
  fontSize: theme('size', {
    medium: '14px',
  }),
  fontWeight: theme('size', {
    medium: '400',
  }),
  lineHeight: theme('size', {
    medium: '23px',
  }),
};

export const Display1M = {
  fontName: theme('size', {
    medium: "'Red Hat Text', sans-serif",
  }),
  fontSize: theme('size', {
    medium: '14px',
  }),
  fontWeight: theme('size', {
    medium: '500',
  }),
  lineHeight: theme('size', {
    medium: '23px',
  }),
};

export const Display2 = {
  fontName: theme('size', {
    medium: "'Red Hat Text', sans-serif",
  }),
  fontSize: theme('size', {
    medium: '13px',
  }),
  fontWeight: theme('size', {
    medium: '400',
  }),
  lineHeight: theme('size', {
    medium: '21px',
  }),
};
export const Display3 = {
  fontName: theme('size', {
    medium: "'Red Hat Text', sans-serif",
  }),
  fontSize: theme('size', {
    medium: '13px',
  }),
  fontWeight: theme('size', {
    medium: '500',
  }),
  lineHeight: theme('size', {
    medium: '21px',
  }),
};
export const Display4 = {
  fontName: theme('size', {
    medium: "'Red Hat Text', sans-serif",
  }),
  fontSize: theme('size', {
    medium: '12px',
  }),
  fontWeight: theme('size', {
    medium: '400',
  }),
  lineHeight: theme('size', {
    medium: '20px',
  }),
};
export const Display5 = {
  fontName: theme('size', {
    medium: "'Red Hat Text', sans-serif",
  }),
  fontSize: theme('size', {
    medium: '11px',
  }),
  fontWeight: theme('size', {
    medium: '400',
  }),
  lineHeight: theme('size', {
    medium: '18px',
  }),
};
export const Display6 = {
  fontName: theme('size', {
    medium: "'Red Hat Text', sans-serif",
  }),
  fontSize: theme('size', {
    medium: '11px',
  }),
  fontWeight: theme('size', {
    medium: '500',
  }),
  lineHeight: theme('size', {
    medium: '18px',
  }),
};
