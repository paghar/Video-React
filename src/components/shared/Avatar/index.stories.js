import React from 'react';
import { action } from '@storybook/addon-actions';
import Story from '.';
import { Display3 } from 'theme/fonts';
import image from 'assets/image/christopher-campbell-rDEOVtE7vOs-unsplash.png';

export default { title: 'Avatar' };

export const Avatar = () => (
  <Story
    backgroundColor="#5c4dff"
    imgUrl={image}
    width="31px"
    height="31px"
    borderRadius="100%"
    font={Display3}
    color="#ffffff"
    avatarClick={action('avatarClick')}
  >
    PZ
  </Story>
);

export const NoimageAvatar = () => (
  <Story
    backgroundColor="#5c4dff"
    width="33px"
    height="33px"
    borderRadius="100%"
    font={Display3}
    color="#ffffff"
  >
    PZ
  </Story>
);

export const WithBorder = () => (
  <Story
    backgroundColor="#5c4dff"
    isBorder
    width="31px"
    height="31px"
    borderRadius="100%"
    font={Display3}
    color="#ffffff"
  >
    PZ
  </Story>
);

export const WithImageBorder = () => (
  <Story
    backgroundColor="#5c4dff"
    imgUrl={image}
    isBorder
    width="31px"
    height="31px"
    borderRadius="100%"
    font={Display3}
    color="#ffffff"
  >
    PZ
  </Story>
);

export const differentSize = () => (
  <Story
    backgroundColor="#5c4dff"
    isBorder
    width="100px"
    height="100px"
    borderRadius="100%"
    font={Display3}
    color="#ffffff"
  >
    PZ
  </Story>
);
