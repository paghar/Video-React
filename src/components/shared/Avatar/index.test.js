//@flow
import React from 'react';
import renderer from 'react-test-renderer';
import { Display6 } from 'theme/fonts';
import Avatar from '.';
import { shallow } from 'enzyme';

import image from 'assets/image/christopher-campbell-rDEOVtE7vOs-unsplash.png';

describe('<Avatar />', () => {
  it('Avatar', () => {
    const tree = renderer
      .create(
        <Avatar
          backgroundColor="#7b7b9a"
          imgUrl={image}
          width="53px"
          height="53px"
          borderRadius="100%"
          font={Display6}
          color="#ffffff"
        >
          PZ
        </Avatar>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Avatar Click Event', () => {
    const mockCallBack = jest.fn();
    const wrapper = shallow(
      <Avatar
        backgroundColor="#7b7b9a"
        imgUrl={image}
        width="53px"
        height="53px"
        borderRadius="100%"
        font={Display6}
        color="#ffffff"
        avatarClick={mockCallBack}
      >
        PZ
      </Avatar>
    );
    wrapper.simulate('click');
    expect(mockCallBack).toHaveBeenCalled();
  });

  it('No image Avatar', () => {
    const tree = renderer
      .create(
        <Avatar
          backgroundColor="#7b7b9a"
          width="53px"
          height="53px"
          borderRadius="100%"
          font={Display6}
          color="#ffffff"
        >
          PZ
        </Avatar>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('With border', () => {
    const tree = renderer
      .create(
        <Avatar
          backgroundColor="#7b7b9a"
          isBorder
          width="53px"
          height="53px"
          borderRadius="100%"
          font={Display6}
          color="#ffffff"
        >
          PZ
        </Avatar>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('With image border', () => {
    const tree = renderer
      .create(
        <Avatar
          backgroundColor="#7b7b9a"
          imgUrl={image}
          isBorder
          width="53px"
          height="53px"
          borderRadius="100%"
          font={Display6}
          color="#ffffff"
        >
          PZ
        </Avatar>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('different size', () => {
    const tree = renderer
      .create(
        <Avatar
          backgroundColor="#7b7b9a"
          imgUrl={image}
          isBorder
          width="100px"
          height="100px"
          borderRadius="100%"
          font={Display6}
          color="#ffffff"
        >
          PZ
        </Avatar>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('different radius', () => {
    const tree = renderer
      .create(
        <Avatar
          backgroundColor="#7b7b9a"
          imgUrl={image}
          isBorder
          width="70px"
          height="70px"
          borderRadius="100%"
          font={Display6}
          color="#ffffff"
        >
          PZ
        </Avatar>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
