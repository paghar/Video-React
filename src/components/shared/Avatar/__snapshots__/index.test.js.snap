// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Avatar /> Avatar 1`] = `
.c0 {
  background-color: #7b7b9a;
  background-image: url(christopher-campbell-rDEOVtE7vOs-unsplash.png);
  background-size: 100% 100%;
  width: 53px;
  height: 53px;
  border-radius: 100%;
  border: 0px solid #0d0d1a;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  outline: none;
}

<button
  className="c0"
  height="53px"
  width="53px"
/>
`;

exports[`<Avatar /> No image Avatar 1`] = `
.c0 {
  background-color: #7b7b9a;
  background-size: 100% 100%;
  width: 53px;
  height: 53px;
  border-radius: 100%;
  border: 0px solid #0d0d1a;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  outline: none;
}

.c1 {
  color: #ffffff;
  pointer-events: none;
  -moz-user-select: none;
  -khtml-user-select: none;
  -webkit-user-select: none;
  -ms-user-select: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

<button
  className="c0"
  height="53px"
  width="53px"
>
  <span
    className="c1"
  >
    PZ
  </span>
</button>
`;

exports[`<Avatar /> With border 1`] = `
.c0 {
  background-color: #7b7b9a;
  background-size: 100% 100%;
  width: 53px;
  height: 53px;
  border-radius: 100%;
  border: 2px solid #0d0d1a;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  outline: none;
}

.c1 {
  color: #ffffff;
  pointer-events: none;
  -moz-user-select: none;
  -khtml-user-select: none;
  -webkit-user-select: none;
  -ms-user-select: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

<button
  className="c0"
  height="53px"
  width="53px"
>
  <span
    className="c1"
  >
    PZ
  </span>
</button>
`;

exports[`<Avatar /> With image border 1`] = `
.c0 {
  background-color: #7b7b9a;
  background-image: url(christopher-campbell-rDEOVtE7vOs-unsplash.png);
  background-size: 100% 100%;
  width: 53px;
  height: 53px;
  border-radius: 100%;
  border: 2px solid #0d0d1a;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  outline: none;
}

<button
  className="c0"
  height="53px"
  width="53px"
/>
`;

exports[`<Avatar /> different radius 1`] = `
.c0 {
  background-color: #7b7b9a;
  background-image: url(christopher-campbell-rDEOVtE7vOs-unsplash.png);
  background-size: 100% 100%;
  width: 70px;
  height: 70px;
  border-radius: 100%;
  border: 2px solid #0d0d1a;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  outline: none;
}

<button
  className="c0"
  height="70px"
  width="70px"
/>
`;

exports[`<Avatar /> different size 1`] = `
.c0 {
  background-color: #7b7b9a;
  background-image: url(christopher-campbell-rDEOVtE7vOs-unsplash.png);
  background-size: 100% 100%;
  width: 100px;
  height: 100px;
  border-radius: 100%;
  border: 2px solid #0d0d1a;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  outline: none;
}

<button
  className="c0"
  height="100px"
  width="100px"
/>
`;
